tests/rptest/services/redpanda.py

```diff
@@ -529,6 +545,16 @@ def cloud_storage_bucket(self):
         elif self.cloud_storage_type == CloudStorageType.ABS:
             return self._cloud_storage_azure_container
 
+    def gcp_iam_token(self):
+        try:
+            res = requests.get(
+                "http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token",
+                headers={"Metadata-Flavor": "Google"})
+        except Exception as ex:
+            self.logger.error(f'Error getting IAM token from GCP: {ex}')
+            raise
+        return res.json()["access_token"]
```

Comment:
nit: this could be within the above try block so if the response does not contain a token we can log it?

Reply:
I removed entirely the `try` `except` block. I used `res.raise_for_status()`. is that ok?